plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -311,13 +310,13 @@ func TestGetCallRates(t *testing.T) {
 			spanKinds:        []string{"SPAN_KIND_SERVER"},
 			groupByOperation: true,
 			updateConfig: func(cfg config.Configuration) config.Configuration {
-				cfg.SupportSpanmetricsConnector = true
 				cfg.MetricNamespace = "span_metrics"
 				return cfg
 			},
 			wantName:        "service_operation_call_rate",
 			wantDescription: "calls/sec, grouped by service & operation",
 			wantLabels: map[string]string{
+				"span_name":    "/OrderResult",
```

Comment:
why are these added? It seems like it would change the semantics of the test.

cc @albertteoh

Reply:
In the original code, the test cases `Enable support for spanmetrics connector with a metric namespace` and `Enable support for spanmetrics connector with normalized metric name` did not include` span_name`(operation) in the `wantLabels `map. This test cases have the `groupByOperation` flag set to true, which means the Prometheus queries generated by the metrics system group by both service_name and span_name.

From the debug output, I confirmed that the metrics returned by these queries do indeed include the span_name label. This change ensures that the test cases accurately reflect the actual behavior when grouping. Including span_name in the wantLabels map ensures that the test cases validate the correct grouping and labeling of metrics. This change does not alter the semantics of the test but rather I believe it aligns the expected output with the current behavior.

If there is any scenario where grouping by operation is not intended, we can adjust the groupByOperation flag or the expected labels accordingly. However, for the current configurations, the inclusion of span_name is necessary and correct.

Reply:
Here's the  more debug output:

Test Name: group by service should be reflected in name/description and query group-by
Expected labels: map[service_name:emailservice]
Actual labels: [name:"service_name" value:"emailservice" ]
Remaining expected labels after matching: map[]
Test Name: group by service and operation should be reflected in name/description and query group-by
Expected labels: map[service_name:emailservice span_name:/OrderResult]
Actual labels: [name:"span_name" value:"/OrderResult"  name:"service_name" value:"emailservice" ]
Remaining expected labels after matching: map[]

Test Name: two services and span kinds result in regex 'or' symbol in query
Expected labels: map[service_name:emailservice]
Actual labels: [name:"service_name" value:"emailservice" ]
Remaining expected labels after matching: map[]

Test Name: enable support for spanmetrics connector with a namespace
Expected labels: map[service_name:emailservice span_name:/OrderResult]
Actual labels: [name:"span_name" value:"/OrderResult"  name:"service_name" value:"emailservice" ]
Remaining expected labels after matching: map[]

Test Name: enable support for spanmetrics connector with normalized metric name
Expected labels: map[service_name:emailservice span_name:/OrderResult]
Actual labels: [name:"span_name" value:"/OrderResult"  name:"service_name" value:"emailservice" ]
Remaining expected labels after matching: map[]

PASS
Process 23592 has exited with status 0

Reply:
Yeah, I think this is correct because the original mock spanmetrics connector JSON response (service_datapoint_response.json) didn't contain the `span_name`/`operation` label. 

Now we're using `service_span_name_datapoint_response.json` as the mock data which contains the `span_name` label.